### PR TITLE
Improve contributing guide & docs

### PR DESCRIPTION
**Pull request by bourgeoisor**
This PR:
- Completely revamps and cleans up the contributing guide
- Adds various templates for new samples
- Cleans up the root README
- Adds a template for new PRs

## How to review?
- Take a look at the brand new contributing guide (https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/165155aa33ada94103c2d08ed924bd86254d724d/.github/CONTRIBUTING.md) and provide feedback (errors, room for improvement, unclear text). Note that some URLs are broken since they're pointing to files that are only checked in with this PR (chicken and egg issue).
- Take a look at the added templates and see if those would help when creating new samples (or if they make it more confusing, of if there's any room for improvement).

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,65 +1,68 @@
...
-## Code reviews
-
-All submissions, including submissions by project members, require review. We
-use GitHub pull requests for this purpose. Consult
-[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
-information on using pull requests.
-
-## Community guidelines
+1. Wait for the proposed changes to be accepted by a repo maintainer.
+1. Sign a [Contributor License Agreement](#contributor-license-agreements) if you haven't done so.
```
Comment (NimJay) [inline .github/CONTRIBUTING.md:7]: **Fix:**
```suggestion
1. Sign a [Contributor License Agreement](#contributor-license-agreements-cla) if you haven't done so.
```

File: README.md [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-See the following resources to learn more:
+See the following resources to get started:
 
 - [Google Kubernetes Engine - Guides](https://cloud.google.com/kubernetes-engine/docs/concepts/kubernetes-engine-overview)
 - [Google Kubernetes Engine - Quickstart](https://cloud.google.com/kubernetes-engine/docs/deploy-app-cluster)
 
+Or click on the following button to open the interactive quickstart demo:
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.svg)](https://ssh.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/kubernetes-engine-samples&cloudshell_tutorial=README.md&cloudshell_workspace=quickstarts/hello-app)
+
```
Comment (NimJay) [inline README.md:14]: This "Open in Cloud Shell" button works. 👍

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+## Tasks
+
+<!-- Once the PR has been created, check boxes as appropriate. -->
+
+* [ ] The [contributing guide](/.github/CONTRIBUTING.md) has been read and followed.
+* [ ] The samples added / modified have been fully tested.
+* [ ] Workflow files have been added / modified, if applicable.
+* [ ] Region tags have been properly added, if new samples.
```
Comment (NimJay) [inline .github/pull_request_template.md:16]: *Suggestion:*
I assume some contributors won't know what a "region tag" is.
```suggestion
* [ ] [Region tags](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/discussions/878) have been properly added, if new samples.
```
Feel free to ignore this suggestion or use a different link.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+REPLACE ME
+
+
+## Tasks
+
+<!-- Once the PR has been created, check boxes as appropriate. -->
+
+* [ ] The [contributing guide](/.github/CONTRIBUTING.md) has been read and followed.
+* [ ] The samples added / modified have been fully tested.
```
Comment (NimJay) [inline .github/pull_request_template.md:14]: **Praise:** I really like this checklist item: "The samples ... have been fully tested".
It hold the contributor accountable, by default.

**Review by NimJay — APPROVED** [review]
This looks amazing! Thanks, @bourgeoisor!

Approved. Feel free to merge whenever — my comments a mostly minor issues.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -1,65 +1,68 @@
...
     - Note that in order for the Cloud Build configs to be applied to the
       `google-samples` project, you need to run `terraform init && terraform apply`
       while in that project (a repository admin will do this for you). [[docs](/.github/terraform/README.md)]
     - The images will be of the form `us-docker.pkg.dev/google-samples/containers/gke<image_name>:latest`
+- Packages and dependencies pinned to up-to-date versions.
+  - Note that we have automation in place to update these versions on a weekly-basis.
+- License headers on all source code and manifest files.
+  [[Example](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/.github/new-samples-templates/cloudbuild.yaml#L1-L13)]
+- Region tags surrounding any file or snippets of code that will be embeded in a tutorial.
+  [[Example](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/ai-ml/llm-multiple-gpus/llm-service.yaml#L15-L28)]
```
Comment (NimJay) [inline .github/CONTRIBUTING.md:48]: *Thought:* I was tempted to suggest something like (for all links to the YAML files):
```suggestion
  [[Example](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/ai-ml/llm-multiple-gpus/llm-service.yaml#L15-L28)]
```

But I think absolute URLs makes sense here.

File: .github/pull_request_template.md (hunk 3 above)
Comment (bourgeoisor) [reply .github/pull_request_template.md:16]: Thought about it but I think it's best to keep all source of truth / docs / details in the "contributing.md" file (which the PR template points to). My answer to "what if someone doesn't know what a region tag is" would be "then they haven't read the contributing guide, which is the first checkbox they're supposed to check"

**Comment by bourgeoisor** [comment]
Cleaned up the contributing guide some more.

Will merge as-is, but always open to improvement / tweaks even post-merge!

**Pull request merged**